packages/cmf/src/cmfConnect.js

```diff
@@ -245,6 +246,21 @@ export default function cmfConnect({
 				router: PropTypes.object,
 			};
 			static WrappedComponent = WrappedComponent;
+			static getState = function getState(state, id = 'default') {
+				return state.cmf.components.getIn([getComponentName(WrappedComponent), id], defaultState);
+			};
+			static setStateAction = function setStateAction(state, id = 'default', type) {
```

Comment:
state parameter to be renamed newState i think
also shouldn't those state update operation also support synchronous state update function ? for safe state update operation when relying on previous state ?

Reply:
newState already exists in the upper declaration
I have added the asyn state update support